### PR TITLE
Add new API version for Server AAD Admin

### PR DESCRIPTION
**Pull request by amolagar5**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by lirenhe** [comment]
@amolagar5 , could you take a look at the tool check error? It seems "$ref": "#/definitions/ProxyResource" is not defined.

**Comment by lirenhe** [comment]
@amolagar5 , I still see a lot of check error, fix take a look at them and do the proper fix.  And let me know if you have any questions.

**Comment by amolagar5** [comment]
Thanks @lirenhe  I will take a look and update the request.

**Comment by amolagar5** [comment]
@lirenhe , can you please help me figure out why checks are failing. I keep getting something like this. 
INFO:swaggertosdk.autorest_tools:  Error: '$.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/administrators/{administratorName}"].put.tags' has incompatible values (---

INFO:swaggertosdk.autorest_tools:- ServerAdministrators

INFO:swaggertosdk.autorest_tools:, ---

INFO:swaggertosdk.autorest_tools:- Administrators

INFO:swaggertosdk.autorest_tools:).

**Comment by lirenhe** [comment]
@amolagar, could you first fix the problem in Avocado and ModelValidation check? This may help to solve the problem for autorest.

**Comment by amolagar5** [comment]
@lirenhe thanks Renhe for taking a look, Looks like swaggers for 2014-04-01 and 2018-06-01-preview have to be exactly same for create or update. Any idea why that is needed. ?

**Comment by lirenhe** [comment]
> @lirenhe thanks Renhe for taking a look, Looks like swaggers for 2014-04-01 and 2018-06-01-preview have to be exactly same for create or update. Any idea why that is needed. ?

Not quite sure about this. What item do you think is 'not needed'?

**Comment by lirenhe** [comment]
@amolagar, there are a couple of failures in prettier check, could you also fix the code style issue?

**Comment by amolagar5** [comment]
@lirenhe how do I fix these prettier check failures, it only says check fails. Does not say why and how to fix them. Can you please share more details on what to fix and how to run tool locally or on git to find more information.

**Comment by amolagar5** [comment]
@lirenhe , can you please take a look when possible, I fixed prettier check and now its complaining about prettier check on files I am deleting and also some breaking change.

**Comment by lirenhe** [comment]
> @lirenhe , can you please take a look when possible, I fixed prettier check and now its complaining about prettier check on files I am deleting and also some breaking change.

We will take a look at those errors.

**Comment by amolagar** [comment]
Thanks

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by qiaozha** [comment]
for breakingchange error, the .azure-pipeline-preproduction/package-lock.json file in your branch is still the old version please merge newest Azure:master into your branch. This breakingchange error should be gone.

**Comment by amolagar5** [comment]
Thanks @qiaozha , Merge fixed it, only pretty check is left which is on a file that is being deleted. Can you please review and sign off.

**Comment by qiaozha** [comment]
> Thanks @qiaozha , Merge fixed it, only pretty check is left which is on a file that is being deleted. Can you please review and sign off.

@NullMDR  please take a look at this pretty check error

**Comment by NullMDR** [comment]
Sorry there is a bug in prettier check. I'll try to fix it.

**Comment by NullMDR** [comment]
@qiaozha @amolagar5 Fixed.

**Review by lirenhe — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "administratorName",
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json]: This should have `enum` . See https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AdministratorProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "AdministratorListResult": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json]: For compatibility this should be ServerAdministratorListResult

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/serverAzureADAdministrators.json#L258

**Review by jaredmoo — CHANGES_REQUESTED** [review]

**Review by jaredmoo — APPROVED** [review]

**Comment by amolagar5** [comment]
@lirenhe whom should I ask for merge now. Is it ARM team. ?

File: specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/serverAzureADAdministrators.json [hunk 3]
```diff
@@ -1,314 +0,0 @@
-{
```
Comment (ryansbenson) [inline specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/serverAzureADAdministrators.json]: Have you gone through the deprecation process for these APIs?

https://armwiki.azurewebsites.net/api_contracts/APIDeprecationPolicy.html

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json [hunk 4]
```diff
@@ -0,0 +1,416 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-06-01-preview",
+    "title": "SqlManagementClient",
```
Comment (ryansbenson) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json]: Title should be human readable

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+    },
+    "SqlVirtualMachineInstanceNameParameter": {
+      "name": "sqlVirtualMachineInstanceName",
+      "in": "path",
+      "description": "The name of the SqlVirtualMachineInstance.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "SqlVirtualMachineContainerNameParameter": {
```
Comment (ryansbenson) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json]: unused

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+      "in": "path",
+      "description": "The name of the blob auditing policy.",
+      "required": true,
+      "type": "string",
+      "enum": [
+        "default"
+      ],
+      "x-ms-parameter-location": "method"
+    },
+    "SqlVirtualMachineInstanceNameParameter": {
```
Comment (ryansbenson) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json]: unused

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+    },
+    "SqlVirtualMachineContainerNameParameter": {
+      "name": "sqlVirtualMachineContainerName",
+      "in": "path",
+      "description": "The name of the SqlVirtualMachineContainer.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "VirtualClusterNameParameter": {
```
Comment (ryansbenson) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json]: unused

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+    },
+    "DatabaseNameParameter": {
+      "name": "databaseName",
+      "in": "path",
+      "description": "The name of the database.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "BlobAuditingPolicyNameParameter": {
```
Comment (ryansbenson) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json]: unused

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+  "parameters": {
+    "ServerNameParameter": {
+      "name": "serverName",
+      "in": "path",
+      "description": "The name of the server.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "ManagedInstanceNameParameter": {
```
Comment (ryansbenson) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json]: unused

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json [hunk 10]
```diff
@@ -0,0 +1,416 @@
...
+      "name": "managedInstanceName",
+      "in": "path",
+      "description": "The name of the managed instance.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "DatabaseNameParameter": {
```
Comment (ryansbenson) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/ServerAzureADAdministrators.json]: unused

File: specification/sql/resource-manager/readme.md [hunk 11]
```diff
@@ -340,7 +340,6 @@ input-file:
 - Microsoft.Sql/stable/2014-04-01/importExport.json
 - Microsoft.Sql/stable/2014-04-01/metrics.json
 - Microsoft.Sql/stable/2014-04-01/replicationLinks.json
-- Microsoft.Sql/stable/2014-04-01/serverAzureADAdministrators.json
```
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: Should not remove

File: specification/sql/resource-manager/readme.md [hunk 12]
```diff
@@ -405,7 +404,6 @@ input-file:
...
```
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: Should not remove

**Comment by amolagar5** [comment]
@ryansbenson added old files back, for title. - @jaredmoo  said its generic one for all SQL, unused params are generated by automation and not worth removing manually.

**Review by ryansbenson — APPROVED** [review]
Approved from ARMs side

**Comment by amolagar5** [comment]
@lirenhe , can you please merge. thanks @ryansbenson  for signing off.

**Comment by lirenhe** [comment]
@amolagar5, I am unable to merge it because of the conflicts. Please help to resolve the conflicts.

**Comment by amolagar5** [comment]
@lirenhe , Sure. Merge complete, pls try again.

**Comment by amolagar** [comment]
Thanks Renhe

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Pull request merged**